### PR TITLE
Several Performance Improvements

### LINKED ISSUE #2: Fix canvas widget sizing issue
The widget image should be able to dynamically resize with the browser width and map to the render window. Currently, this isn't working and the image is a fixed width:

### PR DESCRIPTION
**Pull request by akaszynski**
This is an excellent module and I've already made some use of it on a closed source project, but it's speed leaves much to be desired.  I've cleaned up the interface a bit and changed the implementation somewhat to improve the performance.  Namely:
- Use `put_image_data` with the image array rather using `BytesIO` and writing a fake file.  About a 5-10% performance improvement depending on the size of the window and the base render time.
- Access the pixel data directly with `GetRGBACharPixelData`.  I saw a mention of this on SO and investigated it.  Turns out it's way faster than `vtk.vtkWindowToImageFilter`, and I'm seeing an extra 30% performance improvement on VTK 9 and minimal improvement on v8.  It's good enough that we should consider rewriting the screenshot saver within `pyvista` unless I'm missing something.

#### Miscellaneous Changes
- Set the minimum render based on the first render.  Helps to make sure that the quick renders on million point surface mesh don't block mouse movement or make the lag feel too intense.
- Allow wheel zooming.  It's not going to be an issue when people are scrolling through notebooks as they can just move their mouse to the side, and we really should try to duplicate the standard window within our `Canvas`.  Lots of people are comfortable with mouse wheel zoom, and I was really missing this when playing around with it.  It's an option in the init of `ViewInteractiveWidget`
- *Very* minor refactoring
- Use `weakref.ref` instead of a direct reference to the render window to avoid gc issues.
- Scale mouse movement so movement in the canvas corresponds to the render window (Resolve #2)

### CODE REVIEW

**Comment by akaszynski** [comment]
Pinging @banesullivan for his thoughts...

If possible, it would be great to have this released as `0.1.0`.

The biggest performance bottleneck right now is having to use software rendering as there seems to be no way to render off-screen with a GPU.  It doesn't matter at the moment for me as I'm primarily using docker containers on a kubernetes cluster, but in the future, off-screen rendering would be ideal.

I understand there's a big push in Kitware to move towards a lightweight java client, and I think that will work in most situations, but when you have a gigantic model you'd like to render, it's tough to send across 100 MB (or more) of data and expect the client to render it on a potato.  Server side render does still have its niche.

File: ipyvtk_simple/viewer.py [hunk 1]
```diff
@@ -110,25 +128,31 @@ def set_quick_render_delay(self, delay_sec):
...
+            return data[:, :, :-1]
 
     @throttle(0.1)
     def full_render(self):
         try:
             import time
-
-            self.draw_image(self.get_image(force_render=True))
+            tstart = time.time()
+            self.put_image_data(self.get_image(force_render=True))
```
Comment (banesullivan) [inline ipyvtk_simple/viewer.py:153]: Good find!

File: ipyvtk_simple/viewer.py [hunk 2]
```diff
@@ -110,25 +128,31 @@ def set_quick_render_delay(self, delay_sec):
...
-        )
+        return self._fast_image
+
+    @property
+    def _fast_image(self):
+        import vtk.util.numpy_support as nps
+        import vtk
+        arr = vtk.vtkUnsignedCharArray()
+        self.render_window.GetRGBACharPixelData(0, 0, self.width - 1,
+                                                self.height - 1, 0, arr)
```
Comment (banesullivan) [inline ipyvtk_simple/viewer.py:139]: This is awesome 🤯

Comment (banesullivan) [reply ipyvtk_simple/viewer.py:139]: We definitely need to send that over to PyVista

**Review by banesullivan — APPROVED** [review]
This is excellent work! Thanks, @akaszynski!!

**Pull request merged**